examples/cookie-game/src/index.ts

```diff
@@ -0,0 +1,336 @@
+// @ts-ignore
+import { main, get, set, transaction } from "deku_js_interop"
+import { cookieBakerType } from "./state_types"
+import {
+    createCookieBaker, addCookie, addCursor,
+    addGrandma, addFarm, addMine, addFactory, addBank, addTemple,
+    addWizard, addShipment, addAlchemy, addPortal, addTimeMachine,
+    addAntimatter, addPrism, addChanceMaker, addFractal, addJavascript,
+    addIdleverse, addCordex
+} from "./state"
+import { actions } from "./actions"
+
+const printMessageWithSource = (message: string, source: transaction) => {
+    console.log(message);
+    console.log(source);
+}
+
+const saveState = (source: transaction, sourceValue: cookieBakerType) => {
+    printMessageWithSource("Saving state", sourceValue);
+    set(source, sourceValue);
+    console.log("Successfully saved state");
+}
+
+const transition = (tx: transaction) => {
+    // source -> tz1 address
+    // op_hash / tx_hash => BLAKE2B => resolved as string
+    // operation => any
+    const source = tx.source;
+    const operation = tx.operation;
+    console.log("Getting source");
+    const sourceValue = JSON.parse(get(source));
+    const cookieBaker: cookieBakerType = createCookieBaker(
```

Comment:
What about reworking the `createCookieBaker` function to avoid these 30+ lines?
Would be great to do something like:

```typescript
const cookieBaker: cookieBakerType = JSON.parse(get(source));
```
and hide all this complexity inside `createCookieBaker`